feat: BottomNav 컴포넌트 추가

kimsuyeon0916 opened the pull request:
## 변경 사항

- BottomNav 컴포넌트 및 스토리 추가
- userStore 에 유저 정보 전역 관리

## 리뷰 필요

- 바텀시트와 동일하게 `w-full` 대신 `w-screen` 을 이용하였습니다. 테스트는 로컬에서 모바일 뷰로 적용해서 확인하면 정확합니다. 
(스토리북으로는 라우팅 변화가 정확히 보이지 않고, 현재 너비 이슈가 있어서 테스트하실 때는 로컬로 부탁드립니다!)
- 프로필 모달 열린 이후에 scrim 클릭 시에도 모달이 닫히도록 하였습니다.
  - 프로필 모달 배경은 바텀시트에 적용하였던 컬러와 동일합니다.
- userStore 에 유저 정보(`userId`, `userName`, `family`)를 관리합니다. 추후에 필요한 데이터는 추가할 예정입니다. 로그인 구현 이후에 변경될 수 있습니다.
  - 새로고침 시에도 정보가 초기화되지 않고 유지시키기 위해 persist 미들웨어도 함께 적용하였습니다. 로컬 스토리지에 해당 데이터를 같이 저장합니다.
- 라우팅 관련해서 url 및 방식 등 자유롭게 의견 부탁드립니다!🙌🏻

제가 놓친게 있다면 말씀 부탁드립니다!

### 첨부 링크(옵션)

https://github.com/user-attachments/assets/15781f74-4cfd-4a27-b27f-7fe3bffca6a3


close #41

kimsuyeon0916 committed:
feat: bottom sheet 컴포넌트 z-index 수정 (#41)

kimsuyeon0916 committed:
feat: zIndex 내 bottomNav 프로퍼티 추가 및 상수 변경 (#41)

kimsuyeon0916 committed:
feat: User 타입 추가 (#41)

kimsuyeon0916 committed:
feat: useUserStore 추가 (#41)

kimsuyeon0916 committed:
feat: ProfileModal 컴포넌트 추가 (#41)

kimsuyeon0916 committed:
feat: BottomNav 컴포넌트 추가 (#41)

kimsuyeon0916 committed:
feat: BottomNav 컴포넌트 스토리 추가 (#41)

kimsuyeon0916 committed:
feat: BottomNav 적용 (#41)

kimsuyeon0916 committed:
feat: useUserStore 내 defaultState 변경 (#41)

kimsuyeon0916 committed:
feat: BottonNav 컴포넌트 수정 (#41)

- a -> Link 태그로 변경

kimsuyeon0916 committed:
feat: ProfileModal 컴포넌트 수정 (#41)

- a -> Link 태그로 변경
- handleClickProfile 핸들러 정의

kimsuyeon0916 committed:
feat: userStore 에 persist 미들웨어 설정 (#41)

- 라우팅이 되도 정보를 유지하기 위해 적용

kimsuyeon0916 committed:
feat: BottomNav 라우팅 url 변경 (#41)

kimsuyeon0916 committed:
feat: BottomNav 라우팅 설정 (#41)

kimsuyeon0916 committed:
feat: BottonNav 컴포넌트 수정 (#41)

- 홈 라우팅 주소 변경

kimsuyeon0916 committed:
feat: ProfileModal 컴포넌트 수정 (#41)

- ProfileModalProps 타입 정의 및 컴포넌트 props 추가
- handleClickProfile 핸들러 수정

kimsuyeon0916 committed:
feat: 프로필 모달 scrim 클릭 시, 닫기 설정 (#41)

kimsuyeon0916 committed:
refactor: 프로필, 프로필 모달 컴포넌트 내 button 태그 타입 지정 (#41)

KimDongGyun23 commented on src/components/profile/ProfileModal.tsx:
```diff
@@ -0,0 +1,51 @@
+            }
+
+            return (
+              <div className="Headline-B flex-column-align gap-y-[6px] text-white" key={userId}>
```
`Headline-B` 에서 `headline-B`로 변경해야 할 것 같습니다.

KimDongGyun23 commented on src/components/profile/ProfileModal.tsx:
```diff
@@ -0,0 +1,51 @@
+      <button
+        type="button"
+        onClick={handleClickScrim}
+        className={`fixed inset-0 flex h-[calc(100vh-75px)] w-screen items-end overflow-hidden bg-[rgba(15,23,42,0.5)] px-[52px] pb-[62px]`}
```
`justify-content: center`만 추가되어도 PC 대응 가능할 것 같아요!

kimsuyeon0916 replied:
수정 반영하였습니다~!

KimDongGyun23 commented on src/components/bottomNav/BottomNav.tsx:
```diff
@@ -0,0 +1,81 @@
+  const profileTextStyle = isShowing ? 'caption-M text-mint-9' : 'caption-R text-gray-7'
+
+  return (
+    <nav className={`absolute bottom-0 w-screen ${zIndex.bottomNav}`}>
```
`w-screen`보다 `w-full`을 사용한다면 PC에서도 충분히 대응 가능 할 것 같아요.

![image](https://github.com/user-attachments/assets/d3967b93-9b08-4081-9432-b14a01c3b34d)

kimsuyeon0916 replied:
넵~! z-index 수정하였습니다! 또한 layout 기준으로 반영하는 게 맞는 것 같아 더불어 bottom Sheet 도 w-full 로 수정하였습니다~!

KimDongGyun23 commented on src/components/bottomNav/BottomNav.tsx:
```diff
@@ -0,0 +1,81 @@
+                    className={`${profileTextStyle} flex-column-align absolute bottom-0 gap-y-[6px]`}
+                  >
+                    {/* profile floating action button */}
+                    <div className="flex rounded-t-[50%] bg-white p-[8px] pb-0">
```
현재 사용자 이름을 클릭하면, 다음과 같이 보입니다.
![image](https://github.com/user-attachments/assets/c7d97042-7d31-416e-8484-bc8999658cd4)

이 부분에 z-index가 backdrop보다 낮기 때문에 위와 같이 보인다고 생각합니다.

KimDongGyun23 reviewed (approved):
고생하셨습니다~

kimsuyeon0916 committed:
feat: layout 태그 id 지정 (#41)

kimsuyeon0916 committed:
feat: Portal 컴포넌트 container 변경(#41)

- body -> id='layout' 태그

kimsuyeon0916 committed:
feat: 프로필 모달 컴포넌트 변경 (#41)

- fixed -> absolute
- font 유틸리티 클래스 오타 수정
- w-screen -> wfull
- justify-center 설정

kimsuyeon0916 committed:
feat: Bottom Sheet 컴포넌트 변경 (#41)

- 'use client' 추가
- fixed -> absolute 변경
- backdrop -> bottomNav 로 zIndex 변경
- w-screen -> w-full 변경

kimsuyeon0916 committed:
feat: BottomNav 컴포넌트 변경 (#41)

- w-screen -> w-full 변경
- 불필요한 cursor-pointer 설정 제거

kimsuyeon0916 committed:
feat: 프로필 컴포넌트 변경 (#41)

- 불필요한 cursor-pointer 설정 제거
